src/main/java/mate/academy/springboot/datajpa/service/mapper/impl/ProductMapper.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.springboot.datajpa.service.mapper.impl;
+
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.mapper.RequestDtoMapper;
+import mate.academy.springboot.datajpa.service.mapper.ResponseDtoMapper;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper implements RequestDtoMapper<ProductRequestDto, Product>,
+        ResponseDtoMapper<ProductResponseDto, Product> {
+    private final CategoryService categoryService;
+
+    public ProductMapper(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    @Override
+    public ProductResponseDto mapToDto(Product product) {
+        ProductResponseDto dto = new ProductResponseDto();
+        dto.setId(product.getId());
+        dto.setTitle(product.getTitle());
+        dto.setPrice(product.getPrice());
+        dto.setCategoryId(product.getCategory().getId());
+        return dto;
+    }
+
+    @Override
+    public Product mapToModel(ProductRequestDto dto) {
+        Product product = new Product();
+        product.setTitle(dto.getTitle());
+        product.setPrice(dto.getPrice());
+        product.setCategory(categoryService.get(dto.getCategoryId()));
```

Comment:
Let's not use service dependencies in mapper classes. save just id here and get entity from repository in ProductService.save() method.